src/ec_git_vsn.erl

```diff
@@ -49,24 +49,13 @@ collect_default_refcount(Data) ->
     RawRef = os:cmd("git log -n 1 --pretty=format:'%h\n' "),
 
     {Tag, TagVsn} = parse_tags(Data),
-    RawCount =
-        case Tag of
-            undefined ->
-                os:cmd("git rev-list --count HEAD");
-            _ ->
-                get_patch_count(Tag)
-        end,
+    RawCount = get_patch_count(Tag),
```

Comment:
`get_patch_count` runs `git rev-list --count ~s..HEAD`, which will return 0 on a non-existing thing, whereas here the intent is to deal with a case where the tag doesn't exist; `git rev-list --count HEAD` returning a commit count instead.

We need to either match on `fatal: No names found, cannot describe anything.\n` (which is the error git returns) to return `undefined` or pass in a flag like `--always`, which will return a hash if nothing can be found, but that one makes less sense.

Reply:
> will return 0 on a non-existing thing

You mean an `integer` or the "string" `0`. If it's possible `os:cmd/1` returns a non-string, then the doc.'s wrong (I based myself on the doc. and the `dialyzer` return value).

If I understand correctly, though, you want a fix where the code was eventually breaking (?) I can do that, sure.

Reply:
I'm getting 
```
➜ git rev-list --count 1.0.2..HEAD --always
fatal: ambiguous argument '1.0.2..HEAD':
```
for a non-existing tag. Should we just match on `fatal:` instead? Or is that too broad?

Reply:
Also, I don't know if it was clear, but `parse_tags` only returns string, which is why I removed `undefined`.

Reply:
I've pushed a fix with a test. Not sure `collect_default_refcount` needs to be patched too.

Reply:
Prior to this PR, the first call run is `git rev-list --count ~s..HEAD`. When it fails, we see `fatal: ...\n` -- in this case, we should have returned `undefined` as a tag (which your patch now does). Upon that failure, then we should run `git rev-list --count HEAD` (in the code you were removing), where `os:cmd/1` will return `"<commit count>"` as a version.

The change as implemented at the time of my comment would have always returned `"0"` as a command, or returned an error, whereas the raw count should be `git rev-list --count HEAD` for cases where the tag does not exist.

Reply:
I've hopefully pushed a proper correction now (rebased since the pull request was starting to get all over the place). I'm just not too sure about `build_vsn_string`: `RawRef` is always a string, but you might want to protect it against some condition (in which case it'd become `undefined`).